src/main/scala/ir/Statement.scala

```diff
@@ -64,7 +69,8 @@ class DirectCall(var target: Procedure, var condition: Option[Expr], var returnT
   override def acceptVisit(visitor: Visitor): Jump = visitor.visitDirectCall(this)
 }
 
-class IndirectCall(var target: Variable, var condition: Option[Expr], var returnTarget: Option[Block]) extends Jump {
+case class IndirectCall(var target: Variable, var condition: Option[Expr], var returnTarget: Option[Block],
+                   val address: Option[Int] = None) extends Jump {
```

Comment:
Should revert back to being class instead of case class

Reply:
fixed